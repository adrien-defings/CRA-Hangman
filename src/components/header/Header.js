import React from 'react'
import PropTypes from 'prop-types'

import "./header.css"

const Header = ({ navigationMode, onClick }) => {
  return (
    <div id="header">
      <h2>Hangman Game <span role="img" aria-label="hangman logo">📿</span></h2>
      <p className="lead">Create with Create-react-app x Bootstrap</p>

      {navigationMode !== "home" &&
        <div id="navigationLink">
          <span role="img" aria-label="hangman logo">🏠 </span>
          <span className="btn-link" onClick={() => onClick("home")}>Hangman</span>
          <span onClick={() => onClick("home")}> / {navigationMode === "singlePlayer" ? "Single Player" : "Multiplayer"}
          </span>
        </div>
      }

      <hr className="my-4" />
    </div>

  )
}

Header.defaultProps = {
  navigationMode: "home",
}

Header.propTypes = {
  navigationMode: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
}

export default Header