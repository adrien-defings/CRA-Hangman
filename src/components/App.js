import React, { Component } from 'react'

import 'bootstrap/dist/css/bootstrap.css'
import './app.css'

import Header from './header/Header.js'
import Hangman from './hangman/Hangman.js'
import Home from './home/Home.js'

class App extends Component {
  state = {
    navigationMode: "home",
  }

  setNavigationMode = navigationMode => {
    this.setState({
      navigationMode,
    })
  }

  render() {
    const { navigationMode } = this.state
    return (
      <div id="app" className="jumbotron">
        <Header navigationMode={navigationMode} onClick={this.setNavigationMode} />

        {navigationMode === "home" ?
          <Home onClick={this.setNavigationMode} />
          :
          <Hangman navigationMode={navigationMode} />
        }
      </div>
    )
  }

}

export default App
