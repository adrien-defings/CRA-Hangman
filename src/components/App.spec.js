import React from 'react'

import { shallow } from 'enzyme'
import { expect } from 'chai'

import App from './App'
import Header from './header/Header'

describe('<App />', () => {
  const navigationModes = ["singlePlayer", "multiplayer"]

  // Render the component in the beforeEach
  // => Each test can get a new shallow rendered wrapper
  let wrapper
  beforeEach(() => {
    wrapper = shallow(
      <App />
    )
    expect(wrapper).to.matchSnapshot()
  })

  // can be usefull to test precisely the default onClick prop
  // it('should Header renders with good onClick and navigationMode props', () => {
  //   expect(wrapper.find(Home).props.onClick).to.equal(setNavigationMode("home"))
  // })

  it('should App contain an Header component', () => {
    expect(wrapper.find(Header)).to.be.present()
  })

  it('should navigationMode state be updated with associated set function', () => {
    const mode = navigationModes[Math.floor(Math.random() * navigationModes.length)]
    wrapper.instance().setNavigationMode(mode)
    expect(wrapper).to.have.state('navigationMode').to.equal(mode)
  })
})
