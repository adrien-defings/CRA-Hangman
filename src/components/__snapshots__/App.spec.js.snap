// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should App contain an Header component 1`] = `
<div
  className="jumbotron"
  id="app"
>
  <Header
    navigationMode="home"
    onClick={[Function]}
  />
  <Home
    onClick={[Function]}
  />
</div>
`;

exports[`<App /> should navigationMode state be updated with associated set function 1`] = `
<div
  className="jumbotron"
  id="app"
>
  <Header
    navigationMode="home"
    onClick={[Function]}
  />
  <Home
    onClick={[Function]}
  />
</div>
`;
